src/core/layout/scrollable-list/scrollable-list-handlers.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * @module core/layout/scrollable-list
+ * @copyright BBC 2020
+ * @author BBC Children's D+E
+ * @license Apache-2.0 Apache-2.0
+ */
+/* eslint-disable no-console */
+
+import fp from "../../../../lib/lodash/fp/fp.js";
+
+const GRID_NAME = "grid";
+
+const handleIfVisible = (gelButton, scene) => {
+    const panel = gelButton.rexContainer.parent.getTopmostSizer();
+    const safeArea = scene.layout.getSafeArea({}, false);
+    const height = scene.scale.displaySize.height;
+    const topY = height / 2 + safeArea.y + panel.space.top;
+    const bottomY = topY + panel.innerHeight;
+    const mouseY = scene.input.y;
+    if (mouseY >= topY && mouseY <= bottomY) onClickPlaceholder(gelButton);
+};
+
+const onClickPlaceholder = gelButton => console.log(`Clicked ${gelButton.config.id}`);
+
+const updatePanelOnScroll = panel => () =>
+    getPanelItems(panel).map(item => item.children[0].setElementSizeAndPosition());
+
+const getPanelItems = panel => panel.getByName(GRID_NAME, true).getElement("items");
+
+const updatePanelOnFocus = panel => rexLabel => {
+    const visibleBounds = getVisibleRangeBounds(panel);
+    const itemBounds = getItemBounds(panel, rexLabel);
+    fp.cond([
+        [(vb, ib) => ib.lower < vb.lower, (vb, ib) => updateScrollPosition(panel, ib.lower - vb.lower)],
+        [(vb, ib) => ib.upper > vb.upper, (vb, ib) => updateScrollPosition(panel, ib.upper - vb.upper)],
+        [() => true, () => {}],
+    ])(visibleBounds, itemBounds);
```

Comment:
This feels like it could be written a bit more cleanly as a function. I'm not a huge fan of lodash but the 2 conds looks very similar - so could be possible to do the same operation on a conditional, given the 2 params.

Reply:
(feel free to ignore my bias, commenting for potential discussion)

Reply:
08cc562